src/darwin/Framework/CHIP/MTRDevice.mm

```diff
@@ -1914,13 +1971,26 @@ - (void)_performScheduledExpirationCheck
 
 - (BOOL)_attributeDataValue:(NSDictionary *)one isEqualToDataValue:(NSDictionary *)theOther
 {
-    // Attribute data-value dictionaries are equal if type and value are equal
+    // Sanity check for nil cases
+    if (!one && !theOther) {
```

Comment:
They should not be nil - these are just sanity checks. I will add some MTR_LOG_ERROR lines here to indicate it, so we could see it in case it happens.

Reply:
Done in latest commit